earth_enterprise/src/common/performancelogger.cpp

```diff
@@ -26,34 +28,99 @@
 using namespace std;
 using namespace getime;
 
-// Initialize static members of Profiler class
-PerformanceLogger * const PerformanceLogger::_instance = new PerformanceLogger();
+#ifdef LOG_PERFORMANCE
+
+namespace performance_logger {
+
+// make sure static members get initialized
+string PerformanceLogger::timeFileName = "";
+
+void plMutex::Lock(void) {
+  // if this wasn't properly initialized, we'll get an error
+  int err = pthread_mutex_lock(&mutex);
+  assert(!err);
+  (void) err; // Suppress unused variable 'err' warning.
+}
+
+void plMutex::Unlock(void) {
+  // if this wasn't properly initialized, we'll get an error
+  int err = pthread_mutex_unlock(&mutex);
+  assert(!err);
+  (void) err; // Suppress unused variable 'err' warning.
```

Comment:
could avoid altogether by combining `assert(pthread_mutex_unlock(&mutex));`
might be a candidate for inline-ing

Reply:
Yeah, I think the concern is that when asserts are compiled out, that will also compile out the unlock statement.